src/Tasks/ResolveAssemblyReferences/Client/RarClient.cs

```diff
@@ -0,0 +1,67 @@
+﻿using Microsoft.Build.Framework;
+using Microsoft.Build.Shared;
+using Microsoft.Build.Tasks.ResolveAssemblyReferences.Contract;
+using Microsoft.VisualStudio.Threading;
+using StreamJsonRpc;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.IO.Pipes;
+using System.Linq;
+using System.Security.Principal;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Microsoft.Build.Tasks.ResolveAssemblyReferences.Client
+{
+    internal sealed class RarClient : IDisposable
+    {
+        private const int ConnectionTimeout = 30;
+        private readonly IRarBuildEngine _rarBuildEngine;
+        private NamedPipeClientStream _clientStream;
+
+        public RarClient(IRarBuildEngine rarBuildEngine)
+        {
+            _rarBuildEngine = rarBuildEngine;
+        }
+
+
+        internal bool Connect()
+        {
+            var pipeName = _rarBuildEngine.GetRarPipeName();
+            var stream = _rarBuildEngine.GetRarClientStream(pipeName, ConnectionTimeout);
```

Comment:
is this disposable?

Reply:
Yes, disposing is handled in Dispose method of this class.